hazelcast/src/main/java/com/hazelcast/jet/impl/AbstractJobProxy.java

```diff
@@ -73,6 +74,7 @@ public abstract class AbstractJobProxy<C, M> implements Job {
 
     /** Null for normal jobs, non-null for light jobs  */
     protected final M lightJobCoordinator;
+    protected Subject subject;
```

Comment:
Subject should be available only in `JobProxy`. And it requires a better name/docs: it is set only if the proxy is used to submit a job. It is null when the proxy is obtained for existing job.

Reply:
`invokeSubmitJob` is called in `AbstractJobProxy` constructor, so, to propagate the context, we should have `Subject` in `AbstractJobProxy`.

Reply:
we never use that subject in `ClientJobProxy`, do we?

Reply:
`invokeSubmitJob` is different for client and member

Reply:
why introduce a field that will always be null?

Reply:
https://github.com/hazelcast/hazelcast/pull/25408/commits/0a007c815bfcad778e7946789cb6cd06a5f4829b